Mathlib/MeasureTheory/Group/ModularCharacter.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2025 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import Mathlib.MeasureTheory.Function.LocallyIntegrable
+import Mathlib.MeasureTheory.Group.Integral
+import Mathlib.MeasureTheory.Group.Measure
+import Mathlib.Topology.Metrizable.Urysohn
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+
+/-!
+# Modular character of a locally compact group
+
+On a locally compact group, there is a natural character G → ℝ*, which for g : G gives the value
+μ (· * g) / μ, where μ is an (inner regular) Haar measure. This file defines this character, called
+the modular character, and shows that it is multiplicative and independent of the chosen Haar
+measure.
+TODO: Show that the character is continuous.
+
+## Main Declarations
+
+* `modularCharacter_fun`: Given a measure μ, define the modular character associated with this
+  measure, using the (noncomputable) `haarScalarFactor`. The result that this does not depend on the
+  measure chosen (as long as it is inner regular) is `modularCharacter_fun_eq`.
+* `modularCharacter`: The homomorphism G →* ℝ≥0 whose toFun is `modularCharacter_fun`.
+-/
+
+open MeasureTheory
+open scoped NNReal
+
+namespace MeasureTheory
+
+namespace Measure
+
+variable {G : Type*} [TopologicalSpace G] [Group G] [IsTopologicalGroup G]
+  [MeasurableSpace G] [BorelSpace G] [LocallyCompactSpace G]
+
+open Classical
+
+/-- Definition of the modularCharacter map -/
+@[to_additive addModularCharacterFun "Definition of the addModularCharacter map"]
```

Comment:
instead of specifying the name by hand in all the statements, you should say in the `toAdditive` file that is should always transform `modularCharacter`  to `addModularCharacter`

Reply:
Yes. Look for `fixAbbreviation` there.